db/migrations/20201130190100_create_systems_table.sql

```diff
@@ -0,0 +1,144 @@
+-- migrate:up
+create table systems (
+  id binary(16) not null default (uuid_to_bin(uuid(), 1)),
+  user_id binary(16) not null,
+  name varchar(128) not null,
+  definition JSON not null,
+  created_at timestamp not null default current_timestamp,
+  modified_at timestamp not null default current_timestamp on update current_timestamp,
+  primary key (id),
+  unique system_user_name_key (user_id, name),
+  key systems_user_id_key (user_id),
+  foreign key (user_id)
+    references users(id)
+    on delete cascade on update restrict
+) engine=innodb row_format=compressed;
+
+
+-- get system
+create definer = 'select_objects'@'localhost'
+  procedure get_system (auth0id varchar(32), systemid char(36))
+    comment 'Get the definition for a system'
+    reads sql data sql security definer
+  begin
+    declare binid binary(16);
+    declare allowed boolean default false;
+
+    set binid = uuid_to_bin(systemid, 1);
+    set allowed = exists(
+      select 1 from systems where id = binid and user_id = get_user_binid(auth0id));
```

Comment:
Could use `check_users_system` here. Maybe that requires a reordering of the definitions?